src/AllegroFlare/Screens/Transition.cpp

```diff
@@ -0,0 +1,171 @@
+}
+
+
+bool Transition::get_finished()
+{
+   return finished;
+}
+
+
+void Transition::initialize()
+{
+   if (!(al_is_system_installed()))
+      {
+         std::stringstream error_message;
+         error_message << "Transition" << "::" << "initialize" << ": error: " << "guard \"al_is_system_installed()\" not met";
+         throw std::runtime_error(error_message.str());
+      }
+   if (!((!initialized)))
+      {
+         std::stringstream error_message;
+         error_message << "Transition" << "::" << "initialize" << ": error: " << "guard \"(!initialized)\" not met";
+         throw std::runtime_error(error_message.str());
+      }
+   if (!(al_get_current_display()))
+      {
+         std::stringstream error_message;
+         error_message << "Transition" << "::" << "initialize" << ": error: " << "guard \"al_get_current_display()\" not met";
+         throw std::runtime_error(error_message.str());
+      }
+   ALLEGRO_BITMAP *backbuffer = al_get_backbuffer(al_get_current_display());
+
+   target = al_create_sub_bitmap(
+      backbuffer, 0, 0, al_get_bitmap_width(backbuffer), al_get_bitmap_height(backbuffer));
+   initialized = true;
+   return;
+}
+
+void Transition::primary_timer_func()
+{
+   if (!(initialized))
+      {
+         std::stringstream error_message;
+         error_message << "Transition" << "::" << "primary_timer_func" << ": error: " << "guard \"initialized\" not met";
+         throw std::runtime_error(error_message.str());
+      }
+   if (!(transition_fx))
+      {
+         std::stringstream error_message;
+         error_message << "Transition" << "::" << "primary_timer_func" << ": error: " << "guard \"transition_fx\" not met";
+         throw std::runtime_error(error_message.str());
+      }
+   if (to_screen)
+   {
+      //al_clear_to_color(ALLEGRO_COLOR{0, 0, 0, 0});
+      to_screen->primary_timer_func();
+      draw_backbuffer_to_pasteboard_a_bitmap();
+   }
+
+   if (from_screen)
+   {
+      //al_clear_to_color(ALLEGRO_COLOR{0, 0, 0, 0});
+      from_screen->primary_timer_func();
+      draw_backbuffer_to_pasteboard_b_bitmap();
+   }
+
+   transition_fx->update();
+   ALLEGRO_STATE previous_render_state;
+   al_store_state(&previous_render_state, ALLEGRO_STATE_TARGET_BITMAP);
+   al_set_target_bitmap(target);
+   al_clear_to_color(ALLEGRO_COLOR{0.2, 0.2, 0.21, 1.0});
+   transition_fx->render();
+   al_restore_state(&previous_render_state);
+
+   if (!finished && transition_fx->get_finished())
+   {
+      finished = true;
+      emit_completion_event();
+   }
+
+   return;
+}
+
+void Transition::draw_backbuffer_to_pasteboard_a_bitmap()
+{
+   ALLEGRO_BITMAP *source_bitmap = al_get_backbuffer(al_get_current_display());
+   ALLEGRO_STATE previous_state;
+   al_store_state(&previous_state, ALLEGRO_STATE_TARGET_BITMAP);
+
+   al_set_target_bitmap(transition_fx->get_pasteboard_a());
+
+   al_draw_bitmap(source_bitmap, 0, 0, 0);
+
+   al_restore_state(&previous_state);
+   return;
```

Comment:
This is using the backbuffer to read from, which will be slow when the surface is in a non-identity transformation. This should be scrapped and the scree should always render to an `AllegroFlare::RenderSurfaces::*` surface.